File: client/types/task.ts
import { UserResponse, RecurringTask, SubtaskResponse, TaskPriority, TaskStatus, LabelResponse } from '.'

export type TaskEntry = {
  id?: string
  name: string
  listId?: string | null
  dueDate?: string | null | Date
  duration?: string
  priority?: string | null
  status?: string
  labelIds?: string[] | null
  note?: string
  // recurring?: RecurringTask,
}

export type TaskResponse = {
  id: string
  name: string
  listId?: string | null
  dueDate?: string
  priority?: TaskPriority | null
  createdAt?: string
  updatedAt?: string
  note?: string
  status: TaskStatus
  user: UserResponse
  subtasks?: SubtaskResponse[] | []
  labels?: LabelResponse[] | []
  // recurring?: RecurringTask,
}

File: server/Models/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models
{

    public class Task
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid? ListId { get; set; }
        public Guid? ProjectId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Note { get; set; } = string.Empty;
        public DateTime? DueDate { get; set; }
        public TimeSpan? Duration { get; set; }
        public Priority? Priority { get; set; }
        public Status Status { get; set; }
        public List? List { get; set; }
        public User? User { get; set; }
        public List<Subtask>? Subtasks { get; set; }
        public List<Label>? Labels { get; set; }
        public RecurringTask? Recurring { get; set; }
        public Project? Project { get; set; }

    }
}
